ttnn/cpp/ttnn/operations/eltwise/binary/device/binary_composite_op.cpp

```diff
@@ -392,4 +392,26 @@ Tensor _polyval(const Tensor& input_a, const std::vector<float>& coeffs, const s
     return final_tensor;
 }
 
+Tensor ExecuteGCD::operator()(const Tensor& input_a, const Tensor& input_b, const std::optional<MemoryConfig>& output_mem_config) {
+    Tensor val;
+    Tensor tmp_result;
+    Tensor result = ttnn::full_like(input_a, 0);
+    for(int index =1; index <= 100 ; index++)
+    {
+        val = ttnn::full_like(input_a, index);
+        tmp_result = where(ttnn::logical_and(ttnn::eqz(ttnn::remainder(input_a,val), output_mem_config),ttnn::eqz(ttnn::remainder(input_b,val), output_mem_config)), val, 1);
+        result = ttnn::where(ttnn::gt(tmp_result, result),tmp_result, result);
+    }
+    result =  ttnn::where(ttnn::logical_and(ttnn::eqz(input_a),ttnn::eqz(input_b)), ttnn::full_like(input_a, 0), result);
+    return result;
```

Comment:
Questions:
- Why is the constant 100?
- Can you explain how/why this implementation works?

Reply:
@patrickroberts I have updated it in the description.

Reply:
What if you implemented it like this?
```cpp
Tensor a_gt_b = ttnn::gt(input_a, input_b);
Tensor min = ttnn::where(a_gt_b, input_b, input_a);
Tensor max = ttnn::where(a_gt_b, input_a, input_b);
a_gt_b.deallocate();
// https://www.wolframalpha.com/input?i=26th+Primorial+number
// prime(26)# < std::numeric_limits<float>::max()
constexpr std::size_t max_primorial_index = 26;
for (std::size_t iteration = 0; iteration < max_primorial_index; ++iteration) {
    Tensor isz = ttnn::eqz(min);
    Tensor rem = ttnn::remainder(max, ttnn::where(isz, isz, min));
    max = ttnn::where(isz, max, min);
    min = rem;
}
return max;
```
This is based on https://en.wikipedia.org/wiki/Euclidean_algorithm where
```
function gcd(a, b)
    if b == 0
        return a
    else
        return gcd(b, a mod b)
```
is refactored like this:
```
function gcd(a, b)
    for i = 0; i < 26; i += 1
        rem = a % (b == 0 ? 1 : b)
        a = (b == 0 ? a : b)
        b = rem
    return a
```
where 26 is the theoretical maximum number of remainders that would need to be up to the maximum 32-bit floating point value

Reply:
Actually found out that the upper bound is significantly larger than 26. If my math is correct it is 186, but that could be lower if I can show that the loss of precision prevents that bound from being reached

Reply:
After testing the worst case, the maximum number of iterations is actually only 34 for floating point representation, so here's my updated suggestion:
```cpp
Tensor a_gt_b = ttnn::gt(input_a, input_b);
Tensor min = ttnn::where(a_gt_b, input_b, input_a);
Tensor max = ttnn::where(a_gt_b, input_a, input_b);
a_gt_b.deallocate();
// https://en.wikipedia.org/wiki/Lam%C3%A9%27s_theorem
// While 186 is the theoretical maximum iterations for numbers within the floating point range according to Lame's
// theorem, in practice when evaluating gcd of consecutive Fibonacci numbers coerced to floating point, the
// maximum number of iterations reached is only 34 because the lost precision causes the remainder to
// evaluate to 0 after much fewer iterations.
constexpr std::size_t max_iterations = 34;
for (std::size_t iteration = 0; iteration < max_iterations; ++iteration) {
    Tensor isz = ttnn::eqz(min);
    Tensor rem = ttnn::remainder(max, ttnn::where(isz, isz, min));
    max = ttnn::where(isz, max, min);
    min = rem;
}
return max;
```

Reply:
okay . Let me try it and come back to you.